strings/check_anagram.nim

```diff
@@ -0,0 +1,179 @@
+## Check Anagram
```

Comment:
depending or not whether [this suggestion](https://github.com/TheAlgorithms/Nim/pull/1#discussion_r1133207638) is accepted in contribution guidelines you could add a `{.push raises: [].}` at the top of the file. from a quick look the file should still compile (the assert raises - in debug mode - a Defect that is not considered an exception).